### PR TITLE
Save hiScore and write game data in database

### PR DESCRIPTION
**Pull request by Alex-git-code**
Save hiScore and write game data in database

### CODE REVIEW

File: app/src/main/java/org/alexcode/snake/Level1.java [hunk 1]
```diff
@@ -311,7 +318,7 @@ protected void onPause() {
 
     public boolean onKeyDown(int keyCode, KeyEvent event) {
         if (keyCode == KeyEvent.KEYCODE_BACK) {
-            playGroundView.pause();
+            onPause();
```
Comment (andreisamoila74) [inline app/src/main/java/org/alexcode/snake/Level1.java:321]: why did you change this?

File: app/src/main/java/org/alexcode/snake/Level1.java [hunk 2]
```diff
@@ -361,4 +368,69 @@ public void configureDisplay() {
...
+                            Log.d("UPDATE PLAYER HI SCORE", "SQl error");
+                        }
+                    }
+                } else {
+                    Log.d("UPDATE PLAYER HI SCORE", "Error connecting to database");
+                }
+            }
+
+            @Override
+            public void onFailure(Call<ApiResponse> call, Throwable t) {
```
Comment (andreisamoila74) [inline app/src/main/java/org/alexcode/snake/Level1.java:396]: why is this method empty?

File: app/src/main/java/org/alexcode/snake/Level1.java [hunk 3]
```diff
@@ -361,4 +368,69 @@ public void configureDisplay() {
...
+                    if (response.body().getStatus().equals("ok")) {
+                        if (response.body().getResultCode() == 1) {
+                            Log.d("UPDATE PLAYER STATS", "Player " + playerName + " .Game played increased. Hi Score updated " + hiScore);
+                        } else {
+                            Log.d("UPDATE PLAYER STATS", "Player " + playerName + " .Game played increased.");
+                        }
+                    } else if (response.body().getStatus().equals("failed")) {
+                        if (response.body().getResultCode() == 3) {
+                            Log.d("UPDATE PLAYER STATS", "Player " + playerName + " Failed to update Hi Score and Games played");
+                        } else if (response.body().getResultCode() == 4){
```
Comment (andreisamoila74) [inline app/src/main/java/org/alexcode/snake/Level1.java]:
```suggestion
                        } else if (response.body().getResultCode() == 4) {
```

File: app/src/main/java/org/alexcode/snake/Level1.java [hunk 4]
```diff
@@ -361,4 +368,69 @@ public void configureDisplay() {
...
+                            Log.d("UPDATE PLAYER STATS", "Player " + playerName + " Failed to update Games played");
+                        } else if (response.body().getResultCode() == 5){
+                            Log.d("UPDATE PLAYER STATS", "Player " + playerName + " .No player found with this name");
+                        } else if (response.body().getResultCode() == 6){
```
Comment (andreisamoila74) [inline app/src/main/java/org/alexcode/snake/Level1.java]: here too

File: app/src/main/java/org/alexcode/snake/Level1.java [hunk 5]
```diff
@@ -361,4 +368,69 @@ public void configureDisplay() {
...
+                            Log.d("UPDATE PLAYER STATS", "Player " + playerName + " .Game played increased. Hi Score updated " + hiScore);
+                        } else {
+                            Log.d("UPDATE PLAYER STATS", "Player " + playerName + " .Game played increased.");
+                        }
+                    } else if (response.body().getStatus().equals("failed")) {
+                        if (response.body().getResultCode() == 3) {
+                            Log.d("UPDATE PLAYER STATS", "Player " + playerName + " Failed to update Hi Score and Games played");
+                        } else if (response.body().getResultCode() == 4){
+                            Log.d("UPDATE PLAYER STATS", "Player " + playerName + " Failed to update Games played");
+                        } else if (response.body().getResultCode() == 5){
```
Comment (andreisamoila74) [inline app/src/main/java/org/alexcode/snake/Level1.java]: same here as above

File: app/src/main/java/org/alexcode/snake/MainActivity.java [hunk 6]
```diff
@@ -15,7 +16,7 @@
 public class MainActivity extends AppCompatActivity {
     private Button startGame, profile, ranking, settings;
     private String playerName, volume, language, gameMode;
-    private int gamesPlayed, hiScore;
+    int gamesPlayed, hiScore;
```
Comment (andreisamoila74) [inline app/src/main/java/org/alexcode/snake/MainActivity.java]: why don't you need this variables private anymore?

**Review by andreisamoila74 — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/alexcode/snake/Level1.java [hunk 7]
```diff
@@ -361,4 +368,69 @@ public void configureDisplay() {
...
+                if(response.code() == 200) {
+                    if (response.body().getStatus().equals("ok")) {
+                        if (response.body().getResultCode() == 1) {
+                            hiScore = response.body().getHiScore();
+                            Log.d("UPDATE PLAYER HI SCORE", "Player " + playerName + " Hi Score updated " + hiScore);
+                        }
+                    } else if (response.body().getStatus().equals("failed")) {
+                        if (response.body().getResultCode() == 2) {
+                            Log.d("UPDATE PLAYER HI SCORE", "No player name find. Player " + playerName);
+                        } else if (response.body().getResultCode() == 3){
```
Comment (andreisamoila74) [inline app/src/main/java/org/alexcode/snake/Level1.java]:
```suggestion
                        } else if (response.body().getResultCode() == 3) {
```

File: app/src/main/java/org/alexcode/snake/Level1.java (hunk 1 above)
Comment (Alex-git-code) [reply app/src/main/java/org/alexcode/snake/Level1.java:321]: I wanted to be more clear that I used onPause method

File: app/src/main/java/org/alexcode/snake/Level1.java (hunk 2 above)
Comment (Alex-git-code) [reply app/src/main/java/org/alexcode/snake/Level1.java:396]: I will add a log message

File: app/src/main/java/org/alexcode/snake/Level1.java (hunk 3 above)
Comment (Alex-git-code) [reply app/src/main/java/org/alexcode/snake/Level1.java]: ok

File: app/src/main/java/org/alexcode/snake/Level1.java (hunk 4 above)
Comment (Alex-git-code) [reply app/src/main/java/org/alexcode/snake/Level1.java]: ok

File: app/src/main/java/org/alexcode/snake/Level1.java (hunk 5 above)
Comment (Alex-git-code) [reply app/src/main/java/org/alexcode/snake/Level1.java]: ok

File: app/src/main/java/org/alexcode/snake/MainActivity.java (hunk 6 above)
Comment (Alex-git-code) [reply app/src/main/java/org/alexcode/snake/MainActivity.java]: My mistake. No reason, I will put private again. Thx

File: app/src/main/java/org/alexcode/snake/Level1.java (hunk 7 above)
Comment (Alex-git-code) [reply app/src/main/java/org/alexcode/snake/Level1.java]: ok
